src/Umbraco.Cms.ManagementApi/Controllers/ExamineManagement/HasIndexRebuiltExamineManagementController.cs

```diff
@@ -0,0 +1,81 @@
+﻿using Examine;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Umbraco.Cms.Core.Cache;
+using Umbraco.Cms.Infrastructure.Examine;
+using Umbraco.Cms.ManagementApi.Factories;
+using Umbraco.Cms.ManagementApi.Services;
+using Umbraco.Cms.ManagementApi.ViewModels.ExamineManagement;
+
+namespace Umbraco.Cms.ManagementApi.Controllers.ExamineManagement;
+
+[ApiVersion("1.0")]
+public class HasIndexRebuiltExamineManagementController : ExamineManagementControllerBase
+{
+    private readonly IAppPolicyCache _runtimeCache;
+    private readonly IExamineIndexViewModelFactory _examineIndexViewModelFactory;
+    private readonly IExamineManagerService _examineManagerService;
+
+    public HasIndexRebuiltExamineManagementController(
+        AppCaches runtimeCache,
+        IExamineIndexViewModelFactory examineIndexViewModelFactory,
+        IExamineManagerService examineManagerService)
+    {
+        _runtimeCache = runtimeCache.RuntimeCache;
+        _examineIndexViewModelFactory = examineIndexViewModelFactory;
+        _examineManagerService = examineManagerService;
+    }
+
+    /// <summary>
+    ///     Check if the index has been rebuilt
+    /// </summary>
+    /// <param name="indexName"></param>
+    /// <returns></returns>
+    /// <remarks>
+    ///     This is kind of rudimentary since there's no way we can know that the index has rebuilt, we
+    ///     have a listener for the index op complete so we'll just check if that key is no longer there in the runtime cache
+    /// </remarks>
+    [HttpGet("hasIndexRebuilt")]
+    [MapToApiVersion("1.0")]
+    [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(typeof(ExamineIndexViewModel), StatusCodes.Status200OK)]
+    // This endpoint for now will throw errors if the ExamineIndexViewModel ever has providerProperties defined
+    // This is because System.Text.Json cannot serialize dictionary<string, object>
+    // This has been fixed in .NET 7, so this will work when we upgrade: https://github.com/dotnet/runtime/issues/67588
+    public async Task<ActionResult<ExamineIndexViewModel?>> HasIndexRebuilt(string indexName)
```

Comment:
What is the purpose of this endpoint? It feels kind of weirdly specific to a very specific use case, but maybe it needs to be broader.

In principle the `Indexes` endpoint could be sufficient for querying, since we're not likely to have a whole lot of indexes per site and thus the overhead of returning all indexes is negligible. However, it wouldn't hurt to have an endpoint for retrieving a single index - maybe named `Index` . 

I'd also argue that the "is rebuilding" magic with the runtime cache belongs in the view model factory rather than the controller, and that it is relevant info for any index view model, whether one is querying all indexes or a single index.

So to sum up, here is something to consider:

1. Move the "is rebuilding" detection to the view model factory.
2. Rename this endpoint to `Index` and let it work like the `Indexes` endpoint, just for a specific index name.
3. Leave the populator validation to the `RebuildIndex` endpoint.

Reply:
Great input, I've tried moving this logic to the ViewModelFactory, but a bit unsure of how to handle it properly, if the index is actually rebuilding, for now, we're throwing a OperationCancelledException

Reply:
👍 

I left a suggestion [here](https://github.com/umbraco/Umbraco-CMS/pull/12981/files#r979194417) 😄